packages/aws-cdk-lib/aws-ecs/lib/cluster.ts

```diff
@@ -1300,10 +1300,15 @@ export class AsgCapacityProvider extends Construct {
       this.autoScalingGroup.protectNewInstancesFromScaleIn();
     }
 
+    const capacityProviderNameRegex = /^(?!aws|ecs|fargate).+/gm;
     if (props.capacityProviderName) {
-      if (!(/^(?!aws|ecs|fargate).+/gm.test(props.capacityProviderName))) {
+      if (!(capacityProviderNameRegex.test(props.capacityProviderName))) {
         throw new Error(`Invalid Capacity Provider Name: ${props.capacityProviderName}, If a name is specified, it cannot start with aws, ecs, or fargate.`);
       }
+    } else {
+      if (!(capacityProviderNameRegex.test(Stack.of(this).stackName))) {
+        throw new Error(`Invalid Capacity Provider Name: ${Stack.of(this).stackName}, No name was specified, the stack name cannot start with aws, ecs, or fargate.`);
```

Comment:
Should we throw an error here, or create a valid name automatically as suggested in https://github.com/aws/aws-cdk/issues/29151#issuecomment-1954661302 ?

Sometimes it's difficult to rename a stack, so maybe the latter is better. wdyt?

Reply:
I agree it would be better to generate the name but that has backwards compatibility concerns, I think ?

Reply:
I see. In what kind of situation do you think backwards compatibility will become a problem?

Reply:
If we add code that does this and someone redeploys a stack, CDK will generate a name and it will be different than the name already assigned by CFN.

Reply:
But previously it should have resulted in a deployment error, right?

Reply:
@tmokmss I updated this to generate a capacity provider name when the stack name isn't compliant.